esrally/mechanic/supplier.py

```diff
@@ -525,7 +525,20 @@ def _url_for(self, user_defined_key, default_key, mandatory=True):
                 raise exceptions.SystemSetupError("Neither config key [{}] nor [{}] is defined.".format(user_defined_key, default_key))
             else:
                 return None
-        return url_template.replace("{{VERSION}}", self.version)
+        return self._substitute_vars(url_template)
+
+    def _substitute_vars(self, s):
+
+        substitutions = {
+            "{{VERSION}}": self.version,
+            "{{OSNAME}}": sysstats.os_name().lower(),
+            "{{ARCH}}": sysstats.cpu_arch().lower()
+        }
+        r = s
+        for k in substitutions.keys():
```

Comment:
We could replace this with:

```python
for key, replacement in substitutions.items():
    r = r.replace(key, replacement)
```

This extracts the dictionary key-value pair already in the iterator and we don't need to lookup the value separately.